core/templates/domain/admin/admin-backend-api.service.ts

```diff
@@ -140,6 +140,65 @@ export class AdminBackendApiService {
       });
     });
   }
+
+  async flushCacheAsync(): Promise<void> {
+    return this.http.post<void>(
+      AdminPageConstants.ADMIN_MEMORY_CACHE_HANDLER_URL, {}
+    ).toPromise();
+  }
+
+  async clearSearchIndexAsync(): Promise<void> {
+    return this.http.post<void>(
+      AdminPageConstants.ADMIN_HANDLER_URL, {}
+    ).toPromise();
+  }
+
+  async regenerateTopicRelatedOpportunitiesAsync(
+      topicId: string): Promise<void> {
+    return this.http.post<void>(
+      AdminPageConstants.ADMIN_HANDLER_URL, {
+        action: 'regenerate_topic_related_opportunities',
+        topic_id: topicId
+      }
+    ).toPromise();
+  }
+
+  async uploadTopicSimilaritiesAsync(data: string): Promise<void> {
+    return this.http.post<void>(
+      AdminPageConstants.ADMIN_HANDLER_URL, {
+        action: 'upload_topic_similarities',
+        data: data,
+      }
+    ).toPromise();
+  }
+
+  async sendDummyMailAsync(): Promise<void> {
+    return this.http.post<void>(
+      AdminPageConstants.ADMIN_SEND_DUMMY_MAIL_HANDLER_URL, {}
+    ).toPromise();
+  }
+
+  async fetchMemoryCacheProfileAsync(): Promise<Object> {
+    return this.http.get<Object>(
```

Comment:
Do not use Object type, please define an interface for the response dict and use it here.